### PR TITLE
[OASIS-1] Bump opentelemetry collector dependencies

### PR DESCRIPTION
**Pull request by songy23**
### What does this PR do?

Bump opentelemetry collector dependencies & adopt breaking changes.

### Describe how to test/QA your changes

Verify OTLP configs work as expected. Then send OTLP traces, metrics and logs to Agent and verify they are successfully ingested to DD.

### CODE REVIEW

**Review by dineshg13 — APPROVED** [review]

File: releasenotes/notes/otlp-v0.97.0-3814dd8.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    The OTLP ingest endpoint now supports the same settings and protocol as the OpenTelemetry Collector OTLP receiver v0.97.0.
```
Comment (buraizu) [inline releasenotes/notes/otlp-v0.97.0-3814dd8.yaml:11]:
```suggestion
    The OTLP ingestion endpoint now supports the same settings and protocol as the OpenTelemetry Collector OTLP receiver v0.97.0.
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with a minor suggestion for the release note

**Review by ahmed-mez — APPROVED** [review]

**Review by tobz — APPROVED** [review]

**Review by dustmop — APPROVED** [review]
LGTM

**Pull request merged**